manifests/server.pp

```diff
@@ -131,6 +135,7 @@
   validate_string($rootdn)
   validate_string($rootpw)
   validate_absolute_path($directory)
+  validate_string($backend)
```

Comment:
I intentionally didn't because new backends are bound to appear and then we'd need to update that logic. As an alternative could and should we somehow check that back_${backend}.la exists as a sanity check?